src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/JsonFunctionCallUtilities.cs

```diff
@@ -74,179 +103,141 @@ internal static partial class FunctionCallHelpers
-    {
-        // Fall back to the built-in context since in most cases the return value is JsonElement or JsonNode.
-        options ??= FunctionCallHelperContext.Default.Options;
-        options.MakeReadOnly();
-        return JsonSerializer.SerializeToElement(parameters, options.GetTypeInfo(typeof(IDictionary<string, object>)));
-    }
-
-    /// <summary>
-    /// Serializes a .NET function return parameter to a JSON string.
-    /// </summary>
-    /// <param name="result">The result value to be serialized.</param>
-    /// <param name="options">A <see cref="JsonSerializerOptions"/> governing serialization.</param>
-    /// <returns>A JSON encoding of the parameter.</returns>
-    public static string FormatFunctionResultAsJson(object? result, JsonSerializerOptions? options = null)
-    {
-        // Fall back to the built-in context since in most cases the return value is JsonElement or JsonNode.
-        options ??= FunctionCallHelperContext.Default.Options;
-        options.MakeReadOnly();
-        return JsonSerializer.Serialize(result, options.GetTypeInfo(typeof(object)));
-    }
-
-    /// <summary>
-    /// Serializes a .NET function return parameter to a JSON element.
-    /// </summary>
-    /// <param name="result">The result value to be serialized.</param>
-    /// <param name="options">A <see cref="JsonSerializerOptions"/> governing serialization.</param>
-    /// <returns>A JSON encoding of the parameter.</returns>
-    public static JsonElement FormatFunctionResultAsJsonElement(object? result, JsonSerializerOptions? options = null)
-    {
-        // Fall back to the built-in context since in most cases the return value is JsonElement or JsonNode.
-        options ??= FunctionCallHelperContext.Default.Options;
-        options.MakeReadOnly();
-        return JsonSerializer.SerializeToElement(result, options.GetTypeInfo(typeof(object)));
-    }
-
     /// <summary>
     /// Determines a JSON schema for the provided parameter metadata.
     /// </summary>
     /// <param name="parameterMetadata">The parameter metadata from which to infer the schema.</param>
     /// <param name="functionMetadata">The containing function metadata.</param>
-    /// <param name="options">The global <see cref="JsonSerializerOptions"/> governing serialization.</param>
+    /// <param name="serializerOptions">The global <see cref="JsonSerializerOptions"/> governing serialization.</param>
+    /// <param name="inferenceOptions">The options controlling schema inference.</param>
     /// <returns>A JSON schema document encoded as a <see cref="JsonElement"/>.</returns>
     public static JsonElement InferParameterJsonSchema(
         AIFunctionParameterMetadata parameterMetadata,
         AIFunctionMetadata functionMetadata,
-        JsonSerializerOptions? options)
+        JsonSerializerOptions? serializerOptions,
```

Comment:
We don't want serializerOptions to be optional?

Reply:
> The reason why I left it as required but nullable is that the AIFunctionMetadata might not have a JSO specified, in which case this is almost guaranteed to fail.

Why would it be guaranteed to fail? Couldn't we fall back to whatever default we otherwise use in other situations?

Reply:
Any nonstandard parameter type should guarantee that this fails.